typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -308,8 +316,13 @@
      (cond [(not (unbox typed-context?)) ; no-check, don't check
             #'v]
            [else
-            (define ctc (syntax-local-lift-expression
-                         (make-contract-def-rhs #'ty #f #f)))
+            (define existing-ty-sym (gensym))
+            (define new-ty-ctc (syntax-local-lift-expression
```

Comment:
Could we use an id table based on `new-ty-ctc` (or something else) instead of using a gensym?